src/cargo/core/package.rs

```diff
@@ -571,6 +558,66 @@ impl<'cfg> PackageSet<'cfg> {
         Ok(())
     }
 
+    /// Check if there are any dependency packages that do not have any libs.
+    pub(crate) fn no_lib_pkgs(
+        &self,
+        resolve: &Resolve,
+        root_ids: &[PackageId],
+        has_dev_units: HasDevUnits,
+        requested_kinds: &[CompileKind],
+        target_data: &RustcTargetData<'_>,
+        force_all_targets: ForceAllTargets,
+    ) -> CargoResult<Vec<&Package>> {
```

Comment:
afaict - this function never errors. Perhaps it can return just `Vec<&Package>`?

Reply:
Good catch! Thank you!